src/dom.js

```diff
@@ -161,6 +161,38 @@ export function createElementWithAttributes(doc, tagName, attributes) {
   return addAttributesToElement(element, attributes);
 }
 
+/**
+ * Returns true if node is connected (attached).
+ * @param {!Node} node
+ * @return {boolean}
+ * @see https://dom.spec.whatwg.org/#connected
+ */
+export function isConnectedNode(node) {
+  let n = node;
+  do {
+    n = rootNodeFor(n);
+    if (n.host) {
+      n = n.host;
+    } else {
+      break;
+    }
+  } while (true);
+  return (n instanceof Document);
```

Comment:
how bad is this for performance - should we add guidelines of when/whom should use this? (may be add a presubmit review hook?)